### ISSUE TITLE
cli: `cockroach user` and other SQL CLI commands erroneously requires defaultdb to exist

### ISSUE THREAD

**Issue by dbpaul** [cockroachdb/cockroach#40967]
**Describe the problem**
"cockroach user set ...."  requires the  defaultdb database, but the documentation says 
"The postgres and defaultdb databases can be deleted if they are not needed."

**Please describe the issue you observed, and any steps we can take to reproduce it:**
cockroach sql --insecure --execute="create database mydatabase;"
cockroach init --insecure --host=127.0.0.1
cockroach sql --insecure --execute="drop database IF EXISTS defaultdb;"
cockroach sql --insecure --execute="drop database IF EXISTS postgres;"
cockroach user set username --insecure

Error: pq: database "defaultdb" does not exist
Failed running "user set"

**Expected behavior**
user set should not need defaultdb or the documentation should be updated  that defaultdb can not be deleted for insecure(?) cluster

**Additional data / screenshots**
**Environment:**
Build Tag:    v19.1.3
Build Time:   2019/07/08 18:24:39
Distribution: CCL
Platform:     linux amd64 (x86_64-unknown-linux-gnu)
Go Version:   go1.11.6
C Compiler:   gcc 6.3.0
Build SHA-1:  1a14d34eaa22ca5765f338ab4235766f5e293be0
Build Type:   release

**Comment by ricardocrdb** [comment]
Hey @dbpaul 

I believe you just need to set another database before you are set a new user. So basically, you would run:

```
cockroach sql --insecure --execute="CREATE DATABASE mydatabase;"
cockroach sql --insecure --execute="DROP DATABASE IF EXISTS defaultdb;"
cockroach sql --insecure --execute="DROP DATABASE IF EXISTS postgres;"
cockroach sql --insecure --execute="SET DATABASE mydatabase;"
cockroach user set username --insecure
```

Let me know how that works out for you, or if you have any other questions.

**Comment by knz** [comment]
@ricardocrdb I confirm that the steps you provided do not work. This is actually a bug in `cockroach user` which we must fix in 19.1/19.2.

@dbpaul note that `cockroach user` will be deprecated in CockroachDB 19.2 and will be removed in 20.1. Even if we fix this bug you should change your workflow to use the SQL user management statements instead.

In any case thanks for reporting this issue.
